### DISCUSSION TITLE
How do I derive variables from Placeholders?

### DISCUSSION THREAD

**Discussion by san4d** [cargo-generate/cargo-generate#816] [Q&A] [answered]
What's the idiomatic way to represent derived variables? For example, I want to save the result of `{{ val | downcase }}` to a variable so I don't have to repeat the definition in multiple locations, where `val` was defined in a placeholder block.

I checked a few projects with the [cargo-generate](https://github.com/topics/cargo-generate) topic, but I couldn't find an example.

My thought was to set the variables during the `Pre` hook stage with a `Rhai` script.

**Comment by sassman** [comment, marked as answer]
according to the [liquid docs](https://shopify.github.io/liquid/tags/variable/) it should work this way:

```
{% assign my_variable = false %}
```

But I've never tried it myself.
